### PR TITLE
Ft user

### PR DESCRIPTION
**Pull request by graycadeau**
#### What does this PR do
Has API endpoints to allow a user to create an account and log in

#### PT stories
#160282454
#160282466

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
-from flask import Flask
-
-app = Flask(__name__,instance_relative_config=True)
+"""initializes app as a module"""
```
Comment (maxwellgithinji) [inline app/__init__.py]: too much spacing. use one or two space

File: app/__init__.py [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
 
-"""local import"""
+from flask import Flask
 from . import routes
 
-app.config.from_object('config')
\ No newline at end of file
+app = Flask(__name__, instance_relative_config=True)
+
+app.config.from_object('config')
```
Comment (maxwellgithinji) [inline app/__init__.py]: missing final new line

File: app/routes.py [hunk 3]
```diff
@@ -1 +1,53 @@
...
+    password = request.json.get('password')
+
+    email_regex = re.compile(r"\"?([-a-zA-Z0-9.`?{}]+@\w+\.\w+)\"?")
+    if not email_regex.match(email):
+        return jsonify({"message": "Please enter a valid email"}), 400
+
+    if not password:
+        return jsonify({"msg": "Missing password parameter"}), 400
+
+    return jsonify({"message": "Login successful"}), 201
```
Comment (maxwellgithinji) [inline app/routes.py]: missing final new line

**Comment by maxwellgithinji** [comment]
Code looks great but always remember to lint your code

File: app/routes.py [hunk 4]
```diff
@@ -1 +1,53 @@
...
+
+import re
+from flask import request, jsonify
+
+from app.models import User
+from . import app
+
+
+@app.route('/api/v1/auth/signup', methods=['POST'])
+def signup_user():
```
Comment (Dhul-Husni) [inline app/routes.py]: @graycadeau I thought signup is on challenge 3

Comment (graycadeau) [reply app/routes.py]: They are additional features

**Pull request merged**